src/study_groups.py

```
@@ -0,0 +1,670 @@
+import re
+from urllib.parse import urlparse, parse_qs
+
+
+def _required_text(value):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_required_text`

(ANN202)